### PR TITLE
add cpd support

### PR DESCRIPTION
**Pull request by rhagarty**
- update to latest version of ibm-watson API
- removed tone analyzer and added sentiment detection to assistant
- updated readme's to include support for service on CPD
- changed app to handle different types of results from Disco (passages, SDU, plain text)

### CODE REVIEW

File: doc/source/cf.md [hunk 1]
```diff
@@ -25,8 +25,7 @@ This document shows how to run the application using Cloud Foundry on IBM Cloud.
...
     * `wbc-natural-language-understanding-service`
-    * `wbc-tone-analyzer-service`
 
-> NOTE: The dialog has been reduced to work with Lite plans. If you are looking for all of the previous functionality, you can manually import [`full_banking.json`](../../data/conversation/workspaces/full_banking.json). Follow the instructions in [3. Import the Watson Assistant skill](../../README.md#3-import-the-watson-assistant-skill) to import it, delete the Lite version that was automatically uploaded, and restart the app. The app will recognize the skill by name if only one is available.
+> NOTE: The dialog has been reduced to work with Lite plans. If you are looking for all of the previous functionality, you can manually import [`banking_US.json`](../../data/conversation/workspaces/banking_US.json) (or [`banking_IN.json`](../../data/conversation/workspaces/banking_IN.json) for the India version). Follow the instructions in [3. Import the Watson Assistant skill](../../README.md#3-import-the-watson-assistant-skill) to import it, delete the Lite version that was automatically uploaded, and restart the app. The app will recognize the skill by name if only one is available.
```
Comment (markstur) [inline doc/source/cf.md]: This change does not make sense.  It is talking about the how to use the old full_banking.json.

File: README.md [hunk 2]
```diff
@@ -4,25 +4,27 @@
 
 # Create a banking chatbot with FAQ discovery, anger detection and natural language understanding
 
-In this code pattern, we will create a chatbot using Node.js and Watson Assistant. The Assistant flow will be enhanced by using Natural Language Understanding to identify entities and using Tone Analyzer to detect customer emotions. For FAQs, a call to the Discovery service will use passage retrieval to pull answers from a collection of documents.
+In this code pattern, we will create a chatbot using Node.js and Watson Assistant. The Assistant flow will be enhanced by using Natural Language Understanding to identify location entities. For FAQs, a call to the Discovery service will use passage retrieval to pull answers from a collection of documents.
```
Comment (markstur) [inline README.md]: are we going to detect emotions with Assistant or NLU?

File: doc/source/local.md [hunk 3]
```diff
@@ -94,31 +154,129 @@ Edit the `.env` file with the necessary credentials and settings.
...
+#----------------------------------------------------------
+
 # Watson Assistant
-ASSISTANT_APIKEY=zzZzzABCsU8DBrvi123HLZwVyHbRlBFf_97n9O123ABC
-ASSISTANT_URL=https://gateway.watsonplatform.net/assistant/api
+CONVERSATION_AUTH_TYPE=iam
+CONVERSATION_APIKEY=zzZzzABCsU8DBrvi123HLZwVyHbRlBFf_97n9O123ABC
+CONVERSATION_URL=https://gateway.watsonplatform.net/assistant/api
+# Optionally, use a non-default skill by specifying your own workspace ID.
+WORKSPACE_ID=<add_assistant_workspace_id>
```
Comment (markstur) [inline doc/source/local.md]: should change WORKSPACE_ID to SKILL_ID but that can be done whenever.

File: doc/source/openshift.md [hunk 4]
```diff
@@ -8,19 +8,20 @@ You will need a running OpenShift cluster, or OKD cluster. You can provision [Op
...
 
-* [**Watson Assistant**](https://cloud.ibm.com/catalog/services/conversation)
-* [**Watson Discovery**](https://cloud.ibm.com/catalog/services/discovery)
-* [**Watson Tone Analyzer**](https://cloud.ibm.com/catalog/services/tone-analyzer)
-* [**Watson Natural Language Understanding**](https://cloud.ibm.com/catalog/services/natural-language-understanding)
+* **Watson Assistant**
+* **Watson Discovery**
+* **Watson Natural Language Understanding**
+
+Use the instructions found for [Deploying Locally](local.md) to provision your services from IBM Cloud or from an IBM Cloud Pak for Data cluster. Once you have created the services and collected all of the service credential data, return to this page and complete the remaining steps.
```
Comment (markstur) [inline doc/source/openshift.md]: I think it is a bad idea to send then to local.md from here.  Either move the common stuff to the README (or some other common place) or copy it in here so they don't link to some wrong place.

File: doc/source/openshift.md [hunk 5]
```diff
@@ -47,14 +48,35 @@ Use the following links to create the Watson services on IBM Cloud. Copy/paste t
...
+      * `DISCOVERY_URL`
+      * `DISCOVERY_ENVIRONMENT_ID`
+      * `DISCOVERY_COLLECTION_ID`
+      * `NATURAL_LANGUAGE_UNDERSTANDING_APIKEY`
+      * `NATURAL_LANGUAGE_UNDERSTANDING_URL`
+
+    * For IBM Cloud Pak for Data, the required keys are:
+      * `CONVERSATION_AUTH_TYPE` with value `cp4d`
+      * `CONVERSATION_AUTH_URL`
+      * `CONVERSATION_APIKEY`
```
Comment (markstur) [inline doc/source/openshift.md]: oopsie...  this should be user/pass

File: doc/source/openshift.md [hunk 6]
```diff
@@ -47,14 +48,35 @@ Use the following links to create the Watson services on IBM Cloud. Copy/paste t
...
+      * `DISCOVERY_AUTH_TYPE` with value `cp4d`
+      * `DISCOVERY_AUTH_URL`
+      * `DISCOVERY_APIKEY`
+      * `DISCOVERY_URL`
+      * `DISCOVERY_ENVIRONMENT_ID`
+      * `DISCOVERY_COLLECTION_ID`
+      * `NATURAL_LANGUAGE_UNDERSTANDING_AUTH_TYPE` with value `cp4d`
+      * `NATURAL_LANGUAGE_UNDERSTANDING_AUTH_URL`
+      * `NATURAL_LANGUAGE_UNDERSTANDING_APIKEY`
+      * `NATURAL_LANGUAGE_UNDERSTANDING_URL`
```
Comment (markstur) [inline doc/source/openshift.md]: yeah this part is lacking user/pass and also the disable SSL stuff

Comment (markstur) [reply doc/source/openshift.md]: I'd like to see it similar to this one: https://github.com/IBM/watson-speech-translator/blob/master/doc/source/openshift.md

File: env.sample [hunk 7]
```diff
@@ -1,34 +1,111 @@
...
+# IBM Cloud
+#
+# If your services are running on IBM Cloud,
+# uncomment and configure these.
+# Remove or comment out the IBM Cloud Pak for Data sections.
+#----------------------------------------------------------
 
-# Optionally, use a non-default skill by specifying your own workspace ID or name.
+# Watson Assistant
+# CONVERSATION_AUTH_TYPE=iam
```
Comment (markstur) [inline env.sample]: I consider it a regression to go from ASSISTANT back to CONVERSATION.
Simple workaround is here (Python example):  https://github.com/IBM/watson-voice-bot/blob/master/app.py#L125

File: lib/watson-discovery-setup.js [hunk 8]
```diff
@@ -75,44 +80,46 @@ WatsonDiscoverySetup.prototype.findDiscoveryEnvironment = function(params) {
...
+              reuseEnv = false;
             }
           }
         }
         if (validateID) {
           return reject(new Error('Configured DISCOVERY_ENVIRONMENT_ID=' + validateID + ' not found.'));
         } else if (reuseEnv) {
           console.log('Found existing Discovery environment to use: ', reuseEnv);
-          params.environment_id = reuseEnv.environment_id;
+          params.environmentId = reuseEnv.environment_id;
```
Comment (markstur) [inline lib/watson-discovery-setup.js]: The above change broke this.  For some reason the reuseEnv var is now being treated as a boolean and this code won't work anymore.  Probably just undo the above change.  Definitely do not leave this code trying to evaluate true.environment_id

Comment (markstur) [reply lib/watson-discovery-setup.js]: It used to be undefined or a found, not-read-only environment to try to use.

File: test/unit/test.watson-conversation-setup.js [hunk 9]
```diff
@@ -22,43 +22,50 @@ require('dotenv').config({
...
-      lw.yields(null, { workspaces: [{ name: 'other' }] });
+      lw.yields(null, {
+        result: {
+          workspaces: [{ name: 'other' }]
+        }
+      });
+
       const cw = sinon.stub(c, 'createWorkspace');
-      cw.yields(null, WS);
+      // cw.yields(null, WS);
```
Comment (markstur) [inline test/unit/test.watson-conversation-setup.js]: delete the commented out obsolete code

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,3 +1,8 @@
+## Feb 11, 2019
+- Uses the latest Watson SDK
+- Added support for IBM Cloud Pak for Data
+- Changed enging to Node v10.x
```
[the remark this answers is not in the document]
Comment (markstur) [reply CHANGELOG.md]: honestly I did not remember this having a changelog

[the remark this answers is not in the document]
Comment (rhagarty) [reply CHANGELOG.md]: Caught me by surprise too. Will fix typo

File: doc/source/cf.md (hunk 1 above)
Comment (rhagarty) [reply doc/source/cf.md]: I guess I didn't understand the intent of this. Will change back

File: README.md (hunk 2 above)
Comment (rhagarty) [reply README.md]: Assistant. Will add that to paragraph

File: doc/source/local.md (hunk 3 above)
Comment (rhagarty) [reply doc/source/local.md]: will do this next PR

File: doc/source/openshift.md (hunk 5 above)
Comment (rhagarty) [reply doc/source/openshift.md]: good catch

File: doc/source/openshift.md (hunk 4 above)
Comment (rhagarty) [reply doc/source/openshift.md]: good idea - will create a new common section

File: test/unit/test.watson-conversation-setup.js (hunk 9 above)
Comment (rhagarty) [reply test/unit/test.watson-conversation-setup.js]: will do

File: doc/source/openshift.md (hunk 6 above)
Comment (rhagarty) [reply doc/source/openshift.md]: will use your example

File: lib/watson-discovery-setup.js (hunk 8 above)
Comment (rhagarty) [reply lib/watson-discovery-setup.js]: ok - thought this was an issue because tests failed on this. Will take another look

**Pull request merged**